org.eclipse.dartboard/src/org/eclipse/dartboard/pub/PubspecChangeListener.java

```diff
@@ -0,0 +1,74 @@
+package org.eclipse.dartboard.pub;
+
+import static org.eclipse.core.resources.IResourceDelta.ADDED;
+import static org.eclipse.dartboard.Constants.PUBSPEC;
+
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceChangeEvent;
+import org.eclipse.core.resources.IResourceChangeListener;
+import org.eclipse.core.resources.IResourceDelta;
+import org.eclipse.core.resources.IResourceDeltaVisitor;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.dartboard.Constants;
+import org.eclipse.ui.preferences.ScopedPreferenceStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PubspecChangeListener implements IResourceChangeListener {
+
+	private static final Logger LOG = LoggerFactory.getLogger(PubspecChangeListener.class);
+
+	private ScopedPreferenceStore preferences;
+
+	private PubService pub;
+
+	public PubspecChangeListener() {
+		preferences = new ScopedPreferenceStore(InstanceScope.INSTANCE, Constants.PLUGIN_ID);
```

Comment:
It's a duplicate object for the preference store. We need to access the same object in a whole plugin

Reply:
> It's a duplicate object for the preference store. We need to access the same object in a whole plugin

That's not true. If I understand this https://www.vogella.com/tutorials/EclipsePreferences/article.html correctly, a new object should be created every time it is needed. But again, OSGi service would be nice here.